### ISSUE TITLE
bootstrapped-pip: hash mismatch

### ISSUE THREAD

**Issue by akru** [NixOS/nixpkgs#30158]
## Issue description

I can't build python packages because bootstrapped-pip is broken.

```
$ nix-build '<nixpkgs>' -A pythonPackages.bootstrapped-pip                                                               
these derivations will be built:                                                                                                       
  /nix/store/pwcy08lnhq7drjyzsvxzdjgx71cz93xa-__init__.py.drv
  /nix/store/6an1hi5fnslnm68yaqkjdcgr32vfqzim-python2.7-bootstrapped-pip-9.0.1.drv
building path(s) ‘/nix/store/zsmrlq5ihr4bwkmr6bv4nwpgcln6rrg0-__init__.py’                                                             
                                                                                                                                       
trying https://raw.githubusercontent.com/pypa/setuptools/v36.4.0/pkg_resources/__init__.py
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100  102k  100  102k    0     0   102k      0  0:00:01 --:--:--  0:00:01  619k                                                         
output path ‘/nix/store/zsmrlq5ihr4bwkmr6bv4nwpgcln6rrg0-__init__.py’ has sha256 hash ‘02y99jskkp72m4j12qbdc8gwni5fy8vbmzdz0isgf4k6n330qxjh’ when ‘1wdnq3mammk75mifkdmmjx7yhnpydvnvi804na8ym4mj934l2jkv’ was expected                                                          
cannot build derivation ‘/nix/store/6an1hi5fnslnm68yaqkjdcgr32vfqzim-python2.7-bootstrapped-pip-9.0.1.drv’: 1 dependencies couldn't be built
```

### Steps to reproduce

```
$ curl -o __init__.py https://raw.githubusercontent.com/pypa/setuptools/v36.4.0/pkg_resources/__init__.py
$ nix-hash --base32 --flat --type sha256 __init__.py                                                                      
02y99jskkp72m4j12qbdc8gwni5fy8vbmzdz0isgf4k6n330qxjh 
```

But [build instruction](https://github.com/NixOS/nixpkgs/blob/master/pkgs/development/python-modules/bootstrapped-pip/default.nix) expect `1wdnq3mammk75mifkdmmjx7yhnpydvnvi804na8ym4mj934l2jkv`.

## Technical details

* System: `17.09beta1199.a193400216 (Hummingbird)`
* Nix version: `nix-env (Nix) 1.11.15`
* Nixpkgs version: `17.09.1502.19879836d1`
* Sandboxing enabled: `build-use-sandbox = false`

**Comment by FRidh** [comment]
You are correct. I'll have a look at this. This is relevant for master and 17.09.

**Comment by FRidh** [comment]
Fixed on master with 23ad2b2e7abfef7a0efbba684750fe1039dedcf2 and 17.09 with 2231575a7ddd768fbce7d8f0c75926e5f1ee5c95.
